### ISSUE TITLE
Network keyboard input broken?

### ISSUE THREAD

**Issue by sh-dave** [Kode/Kha#793]
I just tried to run the [BlockArena](https://github.com/kha-samples/blockarena) sample and it doesn't seem to work properly.
The blocks are falling like they're supposed to, but when i press and hold the left or right cursor keys,
they only slightly jitter. Reading the code they should move left and right i think.

Tried with whatever version the Kha submodule points to, and also the current [git](https://github.com/Kode/Kha/tree/e129bda284c029878d803935b292191826a4855f).

**Comment by sh-dave** [comment]
Hm, unable to reproduce it now 😞

**Comment by sh-dave** [comment]
Hm, does it maybe have to do with the typing? I always killed the `/build` folder and build from scratch.

Block.hx

- `@replicated public var x: Float = 0;` doesnt work; only left/right jitter
- `@replicated public var x: Float = 0.0;` doesnt work; only left/right jitter
- `@replicated public var x = 0.0;` seems to work; expected left/right movement

It seems sometimes, the EntityMacro fails to generate correct code. This is from the generated `Block.prototype`:

```js
	,_syncId: function() {
		return this.__syncId;
	}
	,__syncId: null
	,_send: function(offset,bytes) {
		return 0;
	}
	,_receive: function(offset,bytes) {
		return 0;
	}
	,_id: function() {
		return this.__id;
	}
	,_size: function() {
		return 0;
	} 
```

vs.

```js
	,_syncId: function() {
		return this.__syncId;
	}
	,__syncId: null
	,_send: function(offset,bytes) {
		bytes.setDouble(offset,this.x);
		bytes.setDouble(offset + 8,this.y);
		bytes.setDouble(offset + 16,this.sx);
		bytes.setDouble(offset + 24,this.sy);
		bytes.setDouble(offset + 32,this.left);
		bytes.setDouble(offset + 40,this.right);
		return 48;
	}
	,_receive: function(offset,bytes) {
		this.x = bytes.getDouble(offset);
		this.y = bytes.getDouble(offset + 8);
		this.sx = bytes.getDouble(offset + 16);
		this.sy = bytes.getDouble(offset + 24);
		this.left = bytes.getDouble(offset + 32);
		this.right = bytes.getDouble(offset + 40);
		return 48;
	}
	,_id: function() {
		return this.__id;
	}
	,_size: function() {
		return 48;
	} 
```

**Comment by RblSb** [comment]
Also after https://github.com/Kode/Kha/pull/398 clients are out of sync if the rendering of one of them stops for a while (hidden tab in html5 due to requestAnimationFrame).

**Comment by sh-dave** [comment]
I'm closing this (should be fixed with #810) and open another one with the remaining sync issue.
